### ISSUE TITLE
Build failure with the current `main`

### ISSUE THREAD

**Issue by qrilka** [thaw-ui/thaw#118]
Using the commit 9216e5e2d228d6888b17d7afac2f4a32c14235be results in build failure of our app:
```
    --> /home/kirill/ws/rust/thaw/thaw/src/modal/mod.rs:36:20
     |
36   |     let on_enter = move |_| {
     |                    ^^^^^^^^
     = note: required for `leptos::Callback<()>` to implement `From<{closure@/home/kirill/ws/rust/thaw/thaw/src/modal/mod.rs:36:20: 36:28}>`
     = note: required for `{closure@/home/kirill/ws/rust/thaw/thaw/src/modal/mod.rs:36:20: 36:28}` to implement `Into<leptos::Callback<()>>`
note: required by a bound in `CSSTransitionPropsBuilder::<T, CF, IV, (__node_ref, __show, __name, (), __on_after_enter, __on_after_leave, __children)>::on_enter`
    --> /home/kirill/ws/rust/thaw/thaw/src/components/css_transition/mod.rs:4:1
     |
4    | #[component]
     | ^^^^^^^^^^^^ required by this bound in `CSSTransitionPropsBuilder::<T, CF, IV, (__node_ref, __show, __name, (), __on_after_enter, __on_after_leave, __children)>::on_enter`
...
9    |     #[prop(optional, into)] on_enter: Option<Callback<()>>,
     |                             -------- required by a bound in this associated function
     = note: this error originates in the derive macro `::leptos::typed_builder_macro::TypedBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(dyn Callsite + 'static): callback::NotRawCallback` is not satisfied in `{closure@/home/kirill/ws/rust/thaw/thaw/src/modal/mod.rs:36:20: 36:28}`
    --> /home/kirill/ws/rust/thaw/thaw/src/modal/mod.rs:56:5
     |
36   |       let on_enter = move |_| {
     |                      -------- within this `{closure@/home/kirill/ws/rust/thaw/thaw/src/modal/mod.rs:36:20: 36:28}`
...
56   | /     view! {
57   | |         <Teleport>
58   | |             <div
59   | |                 class="thaw-modal-container"
...    |
80   | |                     on_enter
     | |                     --------
     | |                     |
     | |                     this tail expression is of type `{closure@mod.rs:36:20}`
     | |                     required by a bound introduced by this call
...    |
114  | |         </Teleport>
115  | |     }
     | |_____^ within `{closure@/home/kirill/ws/rust/thaw/thaw/src/modal/mod.rs:36:20: 36:28}`, the trait `callback::NotRawCallback` is not implemented for `(dyn Callsite + 'static)`
     |
     = note: required because it appears within the type `&'static (dyn Callsite + 'static)`
```
We're using `leptos` with the feature `nightly` (as far as I remember that's where this problem with `NotRawCallback` appears).

**Comment by qrilka** [comment]
BTW shouldn't there be some CI testing out thaw with `nightly`?

**Comment by disbolog** [comment]
Same issue: https://github.com/thaw-ui/thaw/issues/105#issuecomment-1962291340

**Comment by qrilka** [comment]
Right now we use the version from https://github.com/qrilka/thaw/tree/auto_complete-allow-free-input which doesn't appear to have this problem also in principle we could switch to `stable` as a workaround

**Comment by luoxiaozero** [comment]
I fixed this issue in #119 and added CI.

**Comment by qrilka** [comment]
Thanks, seems that at least the build failure is gone
